pytroll_collectors/s3stalker.py

```diff
@@ -93,18 +185,31 @@ def set_last_fetch(timestamp):
     DatetimeHolder.last_fetch = timestamp
 
 
-def publish_new_files(bucket, config):
+def get_last_fetch():
+    """Get the last fetch time."""
+    return DatetimeHolder.last_fetch
+
+
+def create_messages_for_recent_files(bucket, config, time_back):
+    """Create messages for recent files and return."""
+    logger.debug("Create messages for recent files...")
+    logger.debug("time_back = %s", str(time_back))
+    subject = config['subject']
+    pattern = config.get('file_pattern')
+    with sleeper(2.5):
+        set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
+        s3_kwargs = config['s3_kwargs']
+        fs_, files = get_last_files(bucket, pattern=pattern, **s3_kwargs)
+        messages = filelist_unzip_to_messages(fs_, files, subject)
+        # fs_.clear_instance_cache()
+
+    return messages
+
+
+def publish_new_files(bucket, config, time_back):
     """Publish files newly arrived in bucket."""
+    messages = create_messages_for_recent_files(bucket, config, time_back)
     with Publish("s3_stalker") as pub:
```

Comment:
The publisher should be configurable for example not to use nameserver, allow the user to set the publisher port, etc.

Reply:
Could we do that in another PR (or issue) maybe? It was nothing I touched as far as I know/remember?

Reply:
Certainly! Sorry, switching between different things got confused and thought all of the s3stalker stuff was implemented in this PR :man_facepalming: